api/v3/Membership.php

```diff
@@ -42,7 +59,17 @@
  *   API result array.
  */
 function civicrm_api3_membership_delete($params) {
-  return _civicrm_api3_basic_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (isset($params['preserve_contribution'])) {
+    if (CRM_Member_BAO_Membership::del($params['id'], $params['preserve_contribution'])) {
+      return civicrm_api3_create_success(TRUE, $params);
```

Comment:
Per my comments - I'm actually OK with us not supporting this option from the api level at all. I feel like if people using the api want to delete payment information they can call contribution.delete. I expect more people are accidentally deleting wanted data than enjoying the functionality they are getting at the moment